pkg/controllers/cronfederatedhpa/cronfederatedhpa_job.go

```diff
@@ -252,26 +254,30 @@ func (c *ScalingJob) addFailedExecutionHistory(
 		return exists
 	}
 
-	return retry.RetryOnConflict(retry.DefaultRetry, func() (err error) {
-		// If this history not exist, it means the rule is suspended or deleted, so just ignore it.
-		if exists := addFailedHistoryFunc(); !exists {
+	var operationResult controllerutil.OperationResult
+	if err := retry.RetryOnConflict(retry.DefaultRetry, func() (err error) {
+		var exists bool
+		operationResult, err = helper.UpdateStatus(context.Background(), c.client, cronFHPA, func() error {
+			// If this history not exist, it means the rule deleted, so just ignore it.
+			if exists = addFailedHistoryFunc(); !exists {
+				return errors.New("no such rule")
+			}
 			return nil
+		})
+		if exists && err != nil {
```

Comment:
I'm afraid this brings complexity, the `addFailedHistoryFunc()` does more than a mute-func, it also tells if this updation should be ignored.

Can we refactor(split) the original code first? My draft idea is the `addFailedHistoryFunc` takes two coupled tasks:
1. checks if there is a history record
2. mutate the status

Reply:
good point! done.